### PR TITLE
75 install and configure pylint for project

### PR DESCRIPTION
**Pull request by olexiyko**
Created a branch and pushed it to remote repositor

### CODE REVIEW

**Review by VovaD199 — APPROVED** [review]
We have this same code in the previous PR . I will give my confirmation, but next time do not add files to the commit that are not related to your work on the task.

File: forum/forum/settings.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+# Build paths inside the project like this: BASE_DIR / 'subdir'.
+BASE_DIR = Path(__file__).resolve().parent.parent
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/4.1/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = 'django-insecure-nst1kqbjpud9to&qg=ju#4t#-^znc#zhy9lt1cuytk78ismncg'
```
Comment (mehalyna) [inline forum/forum/settings.py:23]: The `SECRET_KEY` is hardcoded, which is a critical security vulnerability. Use environment variables to store the `SECRET_KEY` in production. 
For example:
```
import os
SECRET_KEY = os.getenv('DJANGO_SECRET_KEY', 'development-secret-key')
```
Then, set the `DJANGO_SECRET_KEY` in your environment variables for production.

File: forum/forum/settings.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
```
Comment (mehalyna) [inline forum/forum/settings.py:26]: `DEBUG` is set to `True`. This should always be `False` in production. Use environment variables to toggle `DEBUG` based on the environment:
```
DEBUG = os.getenv('DJANGO_DEBUG', 'False') == 'True'
```

File: forum/forum/settings.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+ALLOWED_HOSTS = []
```
Comment (mehalyna) [inline forum/forum/settings.py:28]: The `ALLOWED_HOSTS` list is empty, which is insecure for production environments. Configure this for your production domain(s). Example:
```
ALLOWED_HOSTS = os.getenv('DJANGO_ALLOWED_HOSTS', 'localhost').split(',')
```

File: forum/forum/settings.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'NAME': 'postgres',
+        'USER': 'postgres',
+        'PASSWORD': 'postgres',
+        'HOST': 'localhost',
+        'PORT': '5432',
+    }
+}
```
Comment (mehalyna) [inline forum/forum/settings.py:93]: Database credentials (NAME, USER, PASSWORD, etc.) are hardcoded, making them visible in version control. Use environment variables to configure database settings:
```
DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.postgresql_psycopg2',
        'NAME': os.getenv('DB_NAME', 'postgres'),
        'USER': os.getenv('DB_USER', 'postgres'),
        'PASSWORD': os.getenv('DB_PASSWORD', 'postgres'),
        'HOST': os.getenv('DB_HOST', 'localhost'),
        'PORT': os.getenv('DB_PORT', '5432'),
    }
}
```

File: forum/forum/settings.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+# https://docs.djangoproject.com/en/4.1/ref/settings/#databases
+
+# DATABASES = {
+#     'default': {
+#         'ENGINE': 'django.db.backends.sqlite3',
+#         'NAME': BASE_DIR / 'db.sqlite3',
+#     }
+# }
+
+DATABASE = "simpledbmamager"
```
Comment (mehalyna) [inline forum/forum/settings.py:83]: The variable `DATABASE = "simpledbmamager"` is unused and potentially confusing.

File: forum/forum/settings.py (hunk 4 above)
Comment (mehalyna) [inline forum/forum/settings.py:93]: Add a database connection check to catch connection issues during startup:
```
import psycopg2
try:
    conn = psycopg2.connect(
        dbname=os.getenv('DB_NAME', 'postgres'),
        user=os.getenv('DB_USER', 'postgres'),
        password=os.getenv('DB_PASSWORD', 'postgres'),
        host=os.getenv('DB_HOST', 'localhost'),
        port=os.getenv('DB_PORT', '5432'),
    )
    conn.close()
except Exception as e:
    print("Database connection error:", e)
``

File: forum/forum/settings.py [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+MIDDLEWARE = [
+    'django.middleware.security.SecurityMiddleware',
+    'django.contrib.sessions.middleware.SessionMiddleware',
+    'django.middleware.common.CommonMiddleware',
+    'django.middleware.csrf.CsrfViewMiddleware',
+    'django.contrib.auth.middleware.AuthenticationMiddleware',
+    'django.contrib.messages.middleware.MessageMiddleware',
+    'django.middleware.clickjacking.XFrameOptionsMiddleware',
+]
```
Comment (mehalyna) [inline forum/forum/settings.py:50]: Security-related middleware like `XFrameOptionsMiddleware` may not be configured for production. Add security middleware and configure secure headers for production:
```
MIDDLEWARE += [
    'django.middleware.security.SecurityMiddleware',
    'whitenoise.middleware.WhiteNoiseMiddleware',  # For serving static files in production
]

SECURE_BROWSER_XSS_FILTER = True
SECURE_CONTENT_TYPE_NOSNIFF = True
SECURE_HSTS_SECONDS = 3600  # Adjust for production
SECURE_SSL_REDIRECT = os.getenv('DJANGO_SECURE_SSL_REDIRECT', 'True') == 'True'
```

File: forum/forum/settings.py
Comment (mehalyna) [inline forum/forum/settings.py:1]: 1) Use a `.env` file to manage environment variables (e.g., using `python-decouple` or `django-environ`).

2) ) Our project will include an API, ensure `django-cors-headers` is installed and configured to handle cross-origin requests securely.

File: forum/forum/urls.py
Comment (mehalyna) [inline forum/forum/urls.py:1]: 1) There's no root-level route (/). Accessing the base URL will return a 404 error. Add a default view to handle the root path (/). For example:
```
from django.http import HttpResponse

def home(request):
    return HttpResponse("Welcome to the Forum Project")

urlpatterns = [
    path('admin/', admin.site.urls),
    path('', home, name='home'),
]
```

Comment (mehalyna) [inline forum/forum/urls.py:1]: 2. No route exists for health checks or debugging purposes. Add a `/health/` route for monitoring purposes:
```
from django.http import JsonResponse

def health_check(request):
    return JsonResponse({"status": "ok"})

urlpatterns = [
    path('admin/', admin.site.urls),
    path('health/', health_check, name='health_check'),
]
```

Comment (mehalyna) [inline forum/forum/urls.py:1]: 3. As we're using DRF, there’s no support for format suffixes (e.g., .json or .XML). Add `format_suffix_patterns` for API endpoints:
```
from rest_framework.urlpatterns import format_suffix_patterns

urlpatterns = [
    path('admin/', admin.site.urls),
    path('api/users/', include('users.api.urls')),
]

urlpatterns = format_suffix_patterns(urlpatterns)
```

File: forum/manage.py
Comment (mehalyna) [inline forum/manage.py:1]: If someone forgets to activate the virtual environment, the script will fail with a confusing error. Check if the virtual environment is active and warn the user if it's not:
```
if not os.getenv('VIRTUAL_ENV'):
    print("Warning: It seems like the virtual environment is not activated.")
```

**Review by mehalyna — CHANGES_REQUESTED** [review]

File: requirements.txt [hunk 7]
```diff
@@ -0,0 +1,9 @@
+asgiref==3.5.2
+Django==4.1
+psycopg2
+pytz==2022.2.1
+sqlparse==0.4.2
+tzdata==2022.2
+djangorestframework
+pylint==3.3.2
+pylint-django
```
Comment (VovaD199) [inline requirements.txt:9]: You must specify the version of all dependencies.
Update this:
- psycopg2
- djangorestframework
- pylint-django

**Review by VovaD199 — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 7 above)
Comment (olexiyko) [reply requirements.txt:9]: I am already updated it in a 
previous pr

File: forum/forum/settings.py [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+# Application definition
+
+INSTALLED_APPS = [
+    'django.contrib.admin',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+]
```
Comment (IraMladanovych02) [inline forum/forum/settings.py:40]: please, add 'rest_framework' to your INSTALLED_APPS setting.

INSTALLED_APPS = [
    ...
    'rest_framework',
]

File: forum/forum/settings.py [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+                'django.template.context_processors.request',
+                'django.contrib.auth.context_processors.auth',
+                'django.contrib.messages.context_processors.messages',
+            ],
+        },
+    },
+]
+
+WSGI_APPLICATION = 'forum.wsgi.application'
+
```
Comment (IraMladanovych02) [inline forum/forum/settings.py:71]: also you can add REST_FRAMEWORK settings to give access only to authenticated users
REST_FRAMEWORK = {
    # Use Django's standard `django.contrib.auth` permissions,
    # or allow read-only access for unauthenticated users.
    'DEFAULT_PERMISSION_CLASSES': [
        'rest_framework.permissions.DjangoModelPermissionsOrAnonReadOnly'
    ]
}

**Review by IraMladanovych02 — APPROVED** [review]

**Review by mehalyna — APPROVED** [review]

**Review by VovaD199 — APPROVED** [review]

**Pull request merged**